src/main/java/net/mcreator/workspace/Workspace.java

```diff
@@ -408,7 +408,7 @@ public static Workspace readFromFS(File workspaceFile, @Nullable Window ui)
 							"Unsupported generator", JOptionPane.WARNING_MESSAGE);
 					GeneratorConfiguration generatorConfiguration = GeneratorSelector.getGeneratorSelector(ui,
 							GeneratorConfiguration.getRecommendedGeneratorForFlavor(Generator.GENERATOR_CACHE.values(),

```

Comment:
Here in the past, newWorkspace was passed as false (meaning we have workspace relation), but now you call it without MCreator argument, which makes it act as it is a new workspace. Please test this PR properly, or we will have to close it as it is not going in a productive way.

Reply:
I wanted to pass MCreator, but it is simply impossible to provide MCreator argument as it is only created after, when the workspace is actually loaded. However, I tested to (open and then to) change between Forge and Fabric and vice-versa and nothing wrong happened. At this step, giving an MCreator value will simply allow having the popup asking if the user wants to continue as it will allow Forge-Fabric features, which is in all cases not very possible at this state to change the workspace type. The only way would be to change it directly in the `.mcreator` file, but even doing this way, MCreator has currently no way to know if the workspace type has been changed when the workspace was closed. The Forge-Fabric generators can only be chosen at two places, when you create the workspace and when you go inside the workspace settings. Do you understand my decision now?

Reply:
You also need to test all other cases, not just your new feature

Reply:
I did a change that may fix it.

Now, please, go through all use cases of GeneratorSelector, not just new features, and mark it as ready for review once you are sure this works and doesn't break anything, please.

Reply:
I just re-retested and everything works as it should. There are only 3 cases GeneratorSelector is used (excluding tests): when we change the generator type in the workspace settings, when we create a new workspace, and when the last used generator is not found (when re-opening a workspace). For the first case, this is the main usage of the feature, so I tested it several times and it works. In the second case, to have an MCreator instance, so in order to enable the possibility to use a Fabric or Forge generator with the other one, the workspace is not created yet, so there can not have an MCreator instance. however, this is not a problem as we have all possibilities. Finally, the third case is similar to the previous one. It is triggered during the creation of the Workspace variable. This means the workspace var is not fully created and so, the MCreator is not declared yet. When we check where `readFromFS` is used, we can see the MCreator var is declared and initialized after the method call using the Workspace var. As there is no passed MCreator var, MCreator does not allow the user to select a Forge generator (in the case of Fabric if used as the last generator) and vice-versa. MCreator will simply list generators of the last workspace type (Fabric generators if we use the last example or Forge if Forge was used as the last generator). This is just the same behaviour as before.

To conclude, everything works as it should work and as I fully tested before.